pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/OperatorStats.java

```diff
@@ -69,6 +69,11 @@ public void endTimer() {
     }
   }
 
+  public void resetStartTime() {
+    _startTimeMs = -1;
+    _executeStopwatch.reset();
+  }
```

Comment:
instead change the endTimer method
```suggestion
  public void endTimer(TransferableBlock block) {
    if (_executeStopwatch.isRunning()) {
      _executeStopwatch.stop();
    }
    if (block.isNoOpBlock() && processingStarted) {
      _startTimeMs = -1;
      _executeStopwatch.reset();
    } else {
      _processStarted = true;
    }
  }
```